.github/workflows/github-actions-demo.yaml

```diff
@@ -0,0 +1,24 @@
+name: GitHub Actions Demo
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+jobs:
+  Sample-GitHub-Actions:
```

Comment:
I've not seen the job names be verbose like this. Maybe you want to update the Workflow `name`?

AWS workflow example - https://github.com/actions/starter-workflows/blob/main/ci/aws.yml#L30-L33
You can see more examples in the starter-workflows repo's ci directory - https://github.com/actions/starter-workflows/tree/main/ci

Reply:
I referred the job name from here: https://docs.github.com/en/actions/quickstart

Reply:
Even though there's no hard requirement on the naming, it makes sense to add job names like `build`, `test` and `deploy`. Because this was just an initial commit, I thought of keeping it verbose. But as the build steps building the server and CLI, it makes more sense to rename the job to `build`.